### PR TITLE
Allow override of `item.action` and `item.longClickAction`

### PR DESCRIPTION
**Pull request by akloeckner**
This caters for ideas as described in #274. If we can always override the `item`'s `action` or `longPressAction` we can make more fancy things. And TileBoard becomes actually more "customizable".

### CODE REVIEW

**Comment by akloeckner** [comment]
PS: We can argue about the option's name. I was struggling between `longClickAction` and `longPressAction`. The former was used in TileBoard before. The latter seems to be more "common language". Another option would be to use `pressAction`.

**Comment by rchl** [comment]
Can you add documentation, making it clear that this applies to all tile types, not only custom.

**Comment by akloeckner** [comment]
Agreed and fixed.

I also fixed the name of `entityLongClick` for consistency.

**Comment by resoai** [comment]
Perhaps entityClick needs to become entityPress?

**Comment by akloeckner** [comment]
I don't know. I'm quite sure, I wouldn't say “I long-clicked a tile“. But for click, it's also `onClick` for example.Where it was `hmPress` for long press. So, maybe “press“ already implies “long“?

But honestly, I don't know what is the correct wording in the app developer world. ;-)

**Comment by resoai** [comment]
I guess it is “click” with mouse and “tap” with a finger.

**Comment by akloeckner** [comment]
For the sake of consistency, shall we just use the words already established by the HTML attributes? 

https://github.com/resoai/TileBoard/blob/6f1e8d76139c7efaca23f66ed61f26faefae4a61/index.html#L455

That would be “click“ for short and “press“ for long. That's how it is in this PR now. For the user, there's no ambiguity, because `action` is clear without any of these words. And `longPressAction` includes “long“ for clarity.

But, as ever, it's your project, so, I'll change it to whatever you consider best.

**Comment by akloeckner** [comment]
> Can you add documentation

Could have a quick look to check if this is really the shortest way to use UI functions in the custom actions? It seems a bit too complicated for my taste...

```
action: function(item, entity) {return this.$scope.openPopupIframe(item, entity);}
```

**Comment by resoai** [comment]
> > Can you add documentation
> 
> Could you have a quick look to check if this is really the shortest way to use UI functions in the custom actions? It seems a bit too complicated for my taste...
> 
> ```
> action: function(item, entity) {return this.$scope.openPopupIframe(item, entity);}
> ```

To the best of my knowledge it is the only way. We never through about necessity to call those functions from here hence the awkwardness. Her his what I did for `openPage`:

https://github.com/resoai/TileBoard/blob/master/scripts/controllers/main.js#L2039

**Comment by akloeckner** [comment]
Ah, I see. But we shouldn't provide such shortcuts for all possible functions. Whoever wants to use whatever function from `$scope` will have to implement their own shortcut then. Thanks for checking! 

Do you want to change some click/tap/press names still, or shall we keep it as is for the time being?

**Comment by resoai** [comment]
I really don’t mind what we have at the moment. Let’s see what our guru @rchl says :)

**Comment by akloeckner** [comment]
:-) thanks.

**Comment by akloeckner** [comment]
As a note and for automated closing, this fixes #229, by allowing to specify an empty
```
action: function(item, entity) {}
```

File: README.md [hunk 1]
```diff
@@ -280,6 +280,18 @@ Tile Object. [Click here for some real life examples](TILE_EXAMPLES.md)
...
+   * You can override the default action for any tile type.
+   * This function will be evaluated, when the user clicks the tile.
+   */
+  action: function(item, entity) {return this.$scope.openPopupIframe(item, entity);}
+
+  /* longPressAction: Define a custom action on long press
+   * You can override the default action for any tile type.
+   * This function will be evaluated, when the user long-presses the tile.
+   */
+  longPressAction: function(item, entity) {return this.$scope.openPopupIframe(item, entity);}
```
Comment (rchl) [inline README.md]: How about `secondaryAction` then, for parity?
That would leave-out semantics of how they are triggered.
Could even have them triggered in different ways on different platforms (long press on desktop is not that common for example). But that would be out of scope of this change of course.

**Comment by rchl** [comment]
The documentation for CUSTOM tile now says:
> The custom tile type allows you to fire javascript commands on click/tap.
which is weird since all tiles can do that now. It could say, for example:

`The custom tile type doesn't have handling for any specific entity types. It can be used to, for example, trigger custom actions on pressing.`

And the code could also include an example for using `secondaryAction` (or whatever we decide to call it).

File: README.md (hunk 1 above)
Comment (resoai) [reply README.md]: secondaryAction it is than!

**Comment by rchl** [comment]
Merged manually as c28106f4925613d3ccaba990ded5806c7a3a9f18 with some minor corrections - updated commit message and move documentation in README a bit lower.

**Comment by akloeckner** [comment]
You're welcome as ever. Happy to contribute.

**Pull request closed without merging**